discovery-infra/node_controlers/qe_vm_controler.py

```diff
@@ -0,0 +1,88 @@
+import os
+import re
+import libvirt
+import utils
+
+from  node_controlers.node_controler import NodeControler
+
+class QeVmController(NodeControler):
+
+    def __init__(self):
+        self.libvirt_connection = libvirt.open('qemu:///system')
+
+    def __del__(self):
+        self.libvirt_connection.close()
+
+    def list_nodes(self):
+        nodes = {}
+        domains = self.libvirt_connection.listAllDomains()
+
+        for domain in domains:
+            domain_name = domain.name()
+            if ("master" in domain_name) or ("worker" in domain_name):
+                domain_state = "running" if domain.isActive() else "shut_off"
+                nodes[domain_name] = domain_state
+
+        return nodes
+    
+    def shutdown_node(self, node_name):
+        node = self.libvirt_connection.lookupByName(node_name)
+
+        node.destroy()
+
+        return 
+
+    def shutdown_all_nodes(self):
+        nodes = self.list_nodes()
+
+        for node in nodes.keys():
+            self.shutdown_node(node)
+
+        return
+
+    def start_node(self, node_name):
+        node = self.libvirt_connection.lookupByName(node_name)
+
+        node.create()
+
+        return 
+
+    def start_all_nodes(self):
+        nodes = self.list_nodes()
+
+        for node in nodes.keys():
+            self.start_node(node)
+
+        return
+
+    def restart_node(self, node_name):
+        node = self.libvirt_connection.lookupByName(node_name)
+
+        node.destroy()
+        node.create()
+
+        return
+
+    def format_node_disk(self, node_name):
+        command = f"qemu-img info /var/lib/libvirt/images/linchpin/{node_name}.qcow2 | grep 'virtual size'"
```

Comment:
Don't we need to start the VM with the ISO?

Reply:
Right. The boot order for the VMs is set to hd then cdrom. In the case of a reset we need to format the hds so the node will boot from the iso